### ISSUE TITLE
Suspicious amount of time spent in `bad_packet_probe`

### ISSUE THREAD

**Issue by mkeeter** [oxidecomputer/opte#458]
I was capturing kernel flamegraphs for a Crucible Downstairs in the Madrid mini-cluster, and noticed that we're spending ~2% of our runtime in `bad_packet_probe`:

<img width="1917" alt="Screenshot 2024-02-14 at 4 45 34 PM" src="https://github.com/oxidecomputer/opte/assets/745333/c0cbc103-d04f-486f-9f09-21f36d7e2e97">

(and another 1% doing other log formatting in `xde_rx`)

This seems like a lot to me!

Attaching to the probe for 10 seconds, I see that it's mostly `UnexpectedProtocol(TCP)` errors.
```
BRM42220081 # dtrace -q -n 'bad-packet { @[(string)arg0, (string)arg3] = count() } tick-10s { exit(0) }'

  unknown    Parse(BadHeader("BadLength { len: 60 }"))                        80
  unknown    Parse(BadHeader("ReadError { error: NotEnoughBytes }"))         120
  unknown    Parse(UnexpectedProtocol(TCP))                                 4308
```

In other runs, I also saw
```
  unknown    Parse(BadHeader("BadLength { len: 72 }"))                         1
  unknown    Parse(UnexpectedEtherType(6C00))                                  1
  unknown    Parse(UnexpectedProtocol(ICMPv6))                                 1
```

This is on BRM42220081 in cubby 14, which is the scrimlet for the Madrid mini-cluster.

**Comment by FelixMcFelix** [comment]
Thanks a lot for digging this up! Repeating some of what we discussed earlier, this seems to be inbound underlay traffic on `cxgbe0` and `cxgbe1`. Ordinarily we expect to only see IPv6 + UDP + Geneve coming in, which is consistent with the parse errors there – OPTE is unhappy to see TCP, and I suspect it's trying to handle DDM solicits/advertisements as though they are actually Geneve (read: poorly).

Rather luckily, both `fmt` are right next to one another:

https://github.com/oxidecomputer/opte/blob/d6959688ce7f44bfd2975cbebc141356a6a11c21/xde/src/xde.rs#L2064-L2065

I'm keen to completely de-`String` this as far as SDTs are concerned. `ParseError::BadHeader(String)` is another bad apple but I need to actually think on that one, and it's not coming up for **every** non-OPTE underlay packet...

**Comment by FelixMcFelix** [comment]
<img width="1453" alt="image" src="https://github.com/oxidecomputer/opte/assets/6815381/5c8165cd-4a3d-4a5c-9cba-c1672d444a3e">

We seem to be guilty of the same crimes in `Port::process_in`, somewhere.

**Comment by FelixMcFelix** [comment]
Closed by #459.
